src/Analyzers/Core/CodeFixes/ConvertTypeOfToNameOf/AbstractConvertTypeOfToNameOfCodeFixProvider.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ConvertTypeOfToNameOf
+{
+    internal abstract class AbstractConvertTypeOfToNameOfCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+           => ImmutableArray.Create(IDEDiagnosticIds.ConvertTypeOfToNameOfDiagnosticId);
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+               context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = editor.OriginalRoot.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                ConvertTypeOfToNameOf(editor, node, semanticModel);
+            }
+        }
+
+        /// <Summary>
+        ///  Method converts typeof(...).Name to nameof(...)
+        /// </Summary>
+        public void ConvertTypeOfToNameOf(SyntaxEditor editor,
+                                                      SyntaxNode nodeToReplace,
+                                                      SemanticModel semanticModel)
+        {
+            var symbolType = GetSymbolType(nodeToReplace, semanticModel);
+
+            if (symbolType != null)
```

Comment:
this indicates a problem.  That's because you've already offered the fix to a user, but now your fix won't do anything.  Either this case needs to be impossible, or you need to check if you can get a symbol *before* offering the fix.

Reply:
I agree I would elide this check